### PR TITLE
Update serialization tests with `PauliWord` order updates

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Pennylane [#5301](https://github.com/PennyLaneAI/pennylane/pull/5301) updates `PauliWord` to reorder differently when using the `@` operator.

**Description of the Change:**
* Update tests in `test_serialize.py` to accommodate the change mentioned above.
* ~Also added tests for `device_vjp=True` with all interfaces. Not related to the above changes, but it's getting quite cumbersome to merge small changes multiple times.~ These were removed and will be added in a future PR.

Not to be merged until Pennylane [#5301](https://github.com/PennyLaneAI/pennylane/pull/5301) is merged.

Note for reviewers: since some CI overrides the `requirements-dev.txt` file and always uses `pennylane` from master, I've skipped CI for now until [the PR](https://github.com/PennyLaneAI/pennylane/pull/5301) needed for CI in this PR to pass are merged to `pennylane`. But I'd like to have approvals here before I merge that PR so that I can merge this PR as soon as possible after the merge as the changes will cause CI failures in other lightning PRs

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: requirements-dev.txt
Comment (mudit2812) [inline requirements-dev.txt:2]:
```suggestion
git+https://github.com/PennyLaneAI/pennylane.git@master
```

Once the Pennylane PR is merged.

**Comment by mudit2812** [comment]
[sc-57962]

**Comment by AmintorDusko** [comment]
@mudit2812, I added `do not merge` label just in case.
Feel free to remove it as soon as https://github.com/PennyLaneAI/pennylane/pull/5301 is merged.

**Comment by mudit2812** [comment]
Since some CI overrides the `requirements-dev.txt` file and always uses `pennylane` from master, I've skipped CI for now until [the PR](https://github.com/PennyLaneAI/pennylane/pull/5301) needed for CI in this PR to pass are merged to `pennylane`. But I'd like to have approvals here before I merge that PR so that I can merge this PR as soon as possible after the merge as the changes will cause CI failures in other lightning PRs

**Review by vincentmr — APPROVED** [review]
Looks good to me, nice to see this change @mudit2812 !

**Review by maliasadi — APPROVED** [review]
Thanks Mudit! No blocker from my side when these tests pass :green_circle: 

https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/8804624819/job/24165412554?pr=688

**Pull request merged**
